src/core/thread.d

```diff
@@ -1972,6 +1972,10 @@ extern (C) void thread_init()
     //       functions to detect the condition and return immediately.
 
     Thread.initLocks();
+    // The Android VM runtime intercepts SIGUSR1 and apparently doesn't allow
+    // its signal handler to run, so swap the two signals on Android, since
+    // thread_resumeHandler does nothing.
+    version( Android ) thread_setGCSignals(SIGUSR2, SIGUSR1);
```

Comment:
The Android VM runtime [intercepts `SIGUSR1` in its signal-catcher thread](https://android.googlesource.com/platform/art/+/master/runtime/signal_catcher.cc#247) and [apparently doesn't run its signal-handler](https://android.googlesource.com/platform/art/+/master/runtime/signal_catcher.cc#214).  This doesn't matter when D executables are run as standalone processes on the Android command-line, but it causes problems when threads are suspended in an Android app that runs D code in a shared library called from the VM runtime, as that's how native code normally has to be run on Android.

When the unit tests for `core.thread` are run in an Android app without this change, they hang at [`pthread_kill` when it tries to suspend a thread](https://github.com/dlang/druntime/blob/master/src/core/thread.d#L2600).  My guess is this is because the VM runtime intercepts `SIGUSR1` and [the thread_suspendHandler](https://github.com/dlang/druntime/blob/master/src/core/thread.d#L431) is never run.  If I swap the two signals, it still resumes fine, presumably because [the thread_resumeHandler does nothing](https://github.com/dlang/druntime/blob/master/src/core/thread.d#L488), so it doesn't matter if it doesn't run.

I've been using Ilya's #1565 also, which works fine.  However, that pull has been stuck in limbo for more than a year, so I'd like to get this in in the meantime.  Whenever that pull is approved, this can be removed.